cmd/dra-example-kubeletplugin/state.go

```diff
@@ -380,3 +400,56 @@ func GetOpaqueDeviceConfigs(
 
 	return resultConfigs, nil
 }
+
+func (s *DeviceState) buildDeviceStatus(res *resourceapi.DeviceRequestAllocationResult) *resourceapply.AllocatedDeviceStatusApplyConfiguration {
+	dn := res.Device
+	deviceInfo := make(map[string]interface{})
+
+	if d, ok := s.allocatable[dn]; ok {
+		if d.Attributes != nil {
+			attributes := d.Attributes
+
+			if uuid, ok := attributes["uuid"]; ok {
+				deviceInfo["uuid"] = uuid
+			}
+			if model, ok := attributes["model"]; ok {
+				deviceInfo["model"] = model
+			}
+			if driverVersion, ok := attributes["driverVersion"]; ok {
+				deviceInfo["driverVersion"] = driverVersion
+			}
+		}
+	}
+
+	jsonBytes, err := json.Marshal(deviceInfo)
+	if err != nil {
+		klog.Errorf("Failed to marshal device data: %v", err)
+		jsonBytes = []byte("{}")
+	}
+	data := runtime.RawExtension{
+		Raw: jsonBytes,
+	}
+	cond := metav1apply.Condition().
+		WithType("Ready").
+		WithStatus(metav1.ConditionTrue).
```

Comment:
Does the NVIDIA driver also only set this condition at the time the claim is allocated, or can it ever change after that? If it can change, I think the example driver should model how to update that. Even if it stays constant, scaffolding out how that can work with something that says "this is where the latest status is determined" would be helpful.

Reply:
Makes sense. I think for setting device conditions, the interesting part isn't the API call to update the condition, but actually calculating that value of the condition asynchronously. I think I'd rather include that (or no condition at all) than a hardcoded condition added during `NodePrepareResources`.